tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/Net/GraphQLCommon.cs

```diff
@@ -127,12 +131,14 @@ internal static void RecordExecutionErrorsIfPresent(Span span, string errorType,
 
             if (errorCount > 0)
             {
-                RecordExecutionErrors(span, errorType, errorCount, ConstructErrorMessage(executionErrors));
+                RecordExecutionErrors(span, errorType, errorCount, ConstructErrorMessageAndEvents(executionErrors, out var errorSpanEvents), errorSpanEvents);
             }
         }
 
-        private static string ConstructErrorMessage(IExecutionErrors executionErrors)
+        private static string ConstructErrorMessageAndEvents(IExecutionErrors executionErrors, out List<SpanEvent> spanEvents)
```

Comment:
Copy/Paste from other comment

nitpicky but I think this would be better off as two functions, one to ConstructErrorMessage another ConstructErrorSpanEvents
I think that separation of concerns will make it easier to maintain/read in the future.

This would mean then that we need to enumerate executionErrors twice, but I'm assuming that the performance hit there would be negligible